src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensSchema.cs

```diff
@@ -20,81 +20,96 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler.SemanticTokens
         /// Core VS classifications, only map a few things to LSP.  The rest we keep as our own standard classification
         /// type names so those continue to work in VS.
         /// </summary>
-        private static readonly SemanticTokensSchema s_vsTokenSchema = new(
-             new Dictionary<string, string>
-             {
-                 [ClassificationTypeNames.Comment] = SemanticTokenTypes.Comment,
-                 [ClassificationTypeNames.Identifier] = SemanticTokenTypes.Variable,
-                 [ClassificationTypeNames.Keyword] = SemanticTokenTypes.Keyword,
-                 [ClassificationTypeNames.NumericLiteral] = SemanticTokenTypes.Number,
-                 [ClassificationTypeNames.Operator] = SemanticTokenTypes.Operator,
-                 [ClassificationTypeNames.StringLiteral] = SemanticTokenTypes.String,
-             });
+        private static readonly ImmutableDictionary<string, string> s_vsDirectTypeMap = new Dictionary<string, string>()
+        {
+            [ClassificationTypeNames.Comment] = SemanticTokenTypes.Comment,
+            [ClassificationTypeNames.Identifier] = SemanticTokenTypes.Variable,
+            [ClassificationTypeNames.Keyword] = SemanticTokenTypes.Keyword,
+            [ClassificationTypeNames.NumericLiteral] = SemanticTokenTypes.Number,
+            [ClassificationTypeNames.Operator] = SemanticTokenTypes.Operator,
+            [ClassificationTypeNames.StringLiteral] = SemanticTokenTypes.String,
+        }.ToImmutableDictionary();
```

Comment:
Thread safety doesnt seem like a requirement down this path, so readonly should be a bit more performant. Feel free to ping me directly if I misunderstood the reason for the switch.

Reply:
Mainly for clarity - validates that no one could be accidentally changing the dictionary later.

Reply:
yeah, i'd prefer ImmutableDicts so we don't ever have to deal with some crazy issue down the line where someone unintentionally mutated.